dashboard/config/routes.rb

```diff
@@ -325,12 +325,15 @@
     collection do
       get :search
     end
+    member do
+      get :show, to: 'programming_expressions#show_by_id'
+    end
   end
 
   resources :programming_environments, only: [:index, :new, :create, :edit, :update, :show], param: 'name' do
     resources :programming_expressions, param: 'programming_expression_key' do
       member do
-        get :show, to: 'programming_expressions#show_by_keys'
+        get :show, to: 'programming_expressions#show'
```

Comment:
I don't think you should need the to part of this anymore

Reply:
Good catch -- I'll remove